service/matching/taskListManager.go

```diff
@@ -77,36 +76,46 @@ type taskListConfig struct {
 	IdleTasklistCheckInterval  func() time.Duration
 	MinTaskThrottlingBurstSize func() int
 	// taskWriter configuration
-	OutstandingTaskAppendsThreshold int
-	MaxTaskBatchSize                int
+	OutstandingTaskAppendsThreshold func() int
+	MaxTaskBatchSize                func() int
 }
 
-func newTaskListConfig(id *taskListID, config *Config) *taskListConfig {
+func newTaskListConfig(id *taskListID, config *Config, domainCache cache.DomainCache) (*taskListConfig, error) {
+	domainEntry, err := domainCache.GetDomainByID(id.domainID)
```

Comment:
@wxing1292 Can you review this to make sure whether it is OK to host DomainCache within Matching Engine?  Another option is to pass in DomainName along with DomainID on all Matching Engine API calls.

Reply:
@samarabbas  the initialization in matching side looks fine